elastic4s-core/src/main/scala/com/sksamuel/elastic4s/requests/searches/queries/RankFeatureQuery.scala

```diff
@@ -0,0 +1,10 @@
+package com.sksamuel.elastic4s.requests.searches.queries
+
+sealed trait FunctionObject
+final case class Saturation(pivot: Option[Int]) extends FunctionObject
+final case class Logarithm(scalingFactor: Int) extends FunctionObject
+final case class Sigmoid(pivot: Int, exponent: Double) extends FunctionObject
+
+case class RankFeatureQuery(field: String,
+                            boost: Option[Double],
+                            functionObject: Option[FunctionObject]) extends Query
```

Comment:
Together with your other suggestion I've managed to avoid the naming issue altogether.
I've added `saturation`, `log`, and `sigmoid` methods that if one is used clear the other parameters. 
The ADT hierarchy is no longer needed now.

Reply:
I've now renamed them to `withSaturation`, `withLog` and `withSigmoid`.